### ISSUE TITLE
Open keyboard animation

### ISSUE THREAD

**Issue by Sundin** [eBay/NMessenger#48]
When opening the keyboard, the height of the tableview containing the messages seems to change without animating. This causes the behaviour seen in the gif below (here with slow animations enabled).

![ezgif com-optimize](https://cloud.githubusercontent.com/assets/2510354/18636827/78f04c9a-7e8a-11e6-9e8e-c33d13b6b1b1.gif)

**Comment by DavidSchechter** [comment]
@Sundin I see the issue you are talking about.
I don't know currently how to fix it. The tableview size is being changed by the iOS when I update a constraint on the input view.
It could be a bug in iOS.
If we move the scrolling to the commotion block of the animation than it goes away, so it could be cause both animation are happing at the same time, and the scrolling is showing the old height while it's updating.
If more people open bugs like this ones, we will look more into it, as this is only visible when recording the screen in slow motion.

A potential fix would be to change this:
```swift
UIView.animate(withDuration: duration,
                                       delay: TimeInterval(0),
                                       options: animationCurve,
                                       animations: { self.view.layoutIfNeeded()
                                        if self.isKeyboardIsShown {
                                            self.messengerView.scrollToLastMessage(true)
                                        }
                },
                                       completion: nil)
        }
```

To have the option of either scrolling during the animation or in the completion block, depending on a flag set during init.

**Comment by Sundin** [comment]
Hello and thanks for your quick reply!

While this effect is more significant in slow motion, it is still present as a disturbing flicker while running the app in real-time (see the gif below).

![ezgif com-resize](https://cloud.githubusercontent.com/assets/2510354/18660801/bf30a320-7f12-11e6-8ad6-a841f733c4ff.gif)

**Comment by scanniza** [comment]
I am also having this issue, does not look very good when the last messages flicker. 
Do you have any idea what could be the issue? I tried looking at it, it feels like it's just an animation missing somewhere? Maybe animate the change of the table view height? @DavidSchechter  🤔

**Comment by DavidSchechter** [comment]
I have a fix for this issue. However, I cannot merge it because of issues with #42. Once those are resolved I will merged a fix for this issue.

**Comment by DavidSchechter** [comment]
Fixed in #53
